client/starwhale/api/_impl/dataset/builder.py

```diff
@@ -210,271 +242,202 @@ class SWDSBinBuildExecutor(BaseBuildExecutor):
 
     # TODO: add more docstring for class
 
-    _DATA_FMT = SWDS_DATA_FNAME_FMT
-
     class _BinSection(t.NamedTuple):
         offset: int
         size: int
         raw_data_offset: int
         raw_data_size: int
 
-    def _write(self, writer: t.Any, data: bytes) -> _BinSection:
-        size = len(data)
-        crc = crc32(data)  # TODO: crc is right?
-        start = writer.tell()
-        padding_size = self._get_padding_size(size + _header_size)
-
-        _header = _header_struct.pack(
-            _header_magic, crc, 0, size, padding_size, _header_version, _data_magic
-        )
-        _padding = b"\0" * padding_size
-        writer.write(_header + data + _padding)
-        return self._BinSection(
-            offset=start,
-            size=_header_size + size + padding_size,
-            raw_data_offset=start + _header_size,
-            raw_data_size=size,
-        )
-
-    def _get_padding_size(self, size: int) -> int:
-        remain = (size + _header_size) % self.alignment_bytes_size
-        return 0 if remain == 0 else (self.alignment_bytes_size - remain)
-
-    @property
-    def data_format_type(self) -> DataFormatType:
-        return DataFormatType.SWDS_BIN
-
-    def make_swds(self) -> DatasetSummary:
-        # TODO: add lock
-        ds_copy_candidates: t.Dict[str, Path] = {}
-        fno, wrote_size = 0, 0
-
-        dwriter_path = self.data_tmpdir / str(fno)
-        dwriter = dwriter_path.open("wb")
-        ds_copy_candidates[str(fno)] = dwriter_path
-
-        increased_rows = 0
-        total_data_size = 0
-        dataset_annotations: t.Dict[str, t.Any] = {}
-
-        for append_seq_id, item_content in enumerate(
-            self.iter_item(), start=self._forked_last_seq_id + 1
-        ):
-            idx, row_data, row_annotations = self._unpack_row_content(
-                item_content, append_seq_id
+    class _SrcPathSpec(t.NamedTuple):
+        src_path: Path
+        remove_src: bool
+
+    class _BinWriter:
```

Comment:
could we replace the inner class with combination class？ the complex inner class is not good choice. 

in addition, the bin_writer has already bin as an argument.

Reply:
yes sure.

Reply:
done